src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -235,62 +206,85 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * @param details details for validation.
+     * @param targetSrcLine A string with contents of rcurly's line
      * @return true, if right curly should be alone on line
      *         when ALONE_OR_SINGLELINE option is used.
      */
-    private static boolean shouldBeAloneOnLineWithAloneOrSinglelineOption(Details details) {
-        return !isAloneOnLine(details)
+    private static boolean shouldBeAloneOnLineWithAloneOrSinglelineOption(Details details,
+                                                                          String targetSrcLine) {
+        return !isAloneOnLine(details, targetSrcLine)
                 && !isSingleLineBlock(details)
-                && !isAnonInnerClassInit(details.lcurly)
                 && !isEmptyBody(details.lcurly);
     }
 
-    /**
-     * Whether right curly brace starts target source line.
-     * @param details Details of right curly brace for validation
-     * @param targetSourceLine source line to check
-     * @return true if right curly brace starts target source line.
-     */
-    private static boolean isOnStartOfLine(Details details, String targetSourceLine) {
-        return CommonUtil.hasWhitespaceBefore(details.rcurly.getColumnNo(), targetSourceLine)
-                || details.lcurly.getLineNo() == details.rcurly.getLineNo();
-    }
-
     /**
      * Checks whether right curly is alone on a line.
      * @param details for validation.
+     * @param targetSrcLine A string with contents of rcurly's line
      * @return true if right curly is alone on a line.
      */
-    private static boolean isAloneOnLine(Details details) {
+    private static boolean isAloneOnLine(Details details, String targetSrcLine) {
         final DetailAST rcurly = details.rcurly;
-        final DetailAST lcurly = details.lcurly;
         final DetailAST nextToken = details.nextToken;
-        return rcurly.getLineNo() != lcurly.getLineNo()
-            && rcurly.getLineNo() != nextToken.getLineNo();
+        return (rcurly.getLineNo() != nextToken.getLineNo() || skipDoubleBraceInstInit(details))
+                && CommonUtil.hasWhitespaceBefore(details.rcurly.getColumnNo(), targetSrcLine);
+    }
+
+    /**
+     * This method determines if the double brace initialization should be skipped over by the
+     * check. Double brace initializations are treated differently. The corresponding inner
+     * rcurly is treated as if it was alone on line even when it may be followed by another
+     * rcurly and a semi, raising no violations.
+     * <i>Please do note though that the line should not contain anything other than the following
+     * right curly and the semi following it or else violations will be raised.</i>
+     * Only the kind of double brace initializations shown in the following example code will be
+     * skipped over:<br>
+     * <tt>
+     *     Map {@code <String, String>} map = new LinkedHashMap<String, String>() {{<br>
```

Comment:
<>s should be sanitized since this in javadoc.
Also why `tt` instead of `pre`?

Reply:
Sanitized. 

Replace `tt` by `pre`. Used it over `pre` for no particular reason. Also, `pre` is better than `tt` since it preserves whitespaces but `tt` apparently doesn't.